diff --git a/include/target.mk b/include/target.mk
index 92edaec..1f7cc6a 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -201,15 +201,15 @@ ifeq ($(DUMP),1)
     else
       CPU_TYPE ?= mips32
     endif
-    CPU_CFLAGS += -mno-branch-likely
+    CPU_CFLAGS += -mno-branch-likely -mno-mips16 -mno-interlink-compressed -fno-caller-saves -fomit-frame-pointer -mno-long-calls
     CPU_CFLAGS_mips32 = -mips32 -mtune=mips32
     CPU_CFLAGS_mips32r2 = -mips32r2 -mtune=mips32r2
     CPU_CFLAGS_mips64 = -mips64 -mtune=mips64 -mabi=64
-    CPU_CFLAGS_24kc = -mips32r2 -mtune=24kc
-    CPU_CFLAGS_24kec = -mips32r2 -mtune=24kec
-    CPU_CFLAGS_34kc = -mips32r2 -mtune=34kc
-    CPU_CFLAGS_74kc = -mips32r2 -mtune=74kc
-    CPU_CFLAGS_1004kc = -mips32r2 -mtune=1004kc
+    CPU_CFLAGS_24kc = -march=24kc
+    CPU_CFLAGS_24kec = -march=24kec
+    CPU_CFLAGS_34kc = -march=34kc
+    CPU_CFLAGS_74kc = -march=74kc
+    CPU_CFLAGS_1004kc = -march=1004kc
     CPU_CFLAGS_octeon = -march=octeon -mabi=64
     CPU_CFLAGS_dsp = -mdsp
     CPU_CFLAGS_dsp2 = -mdspr2
