diff --git a/toolchain/binutils/Makefile b/toolchain/binutils/Makefile
index c37148e..26d6378 100644
--- a/toolchain/binutils/Makefile
+++ b/toolchain/binutils/Makefile
@@ -18,10 +18,10 @@ ifeq ($(PKG_VERSION),2.25.1)
 endif
 
 ifeq ($(findstring linaro, $(CONFIG_BINUTILS_VERSION)),linaro)
-  PKG_SOURCE_URL:=https://releases.linaro.org/14.09/components/toolchain/binutils-linaro/
-  PKG_REV:=2.24.0-2014.09
+  PKG_SOURCE_URL:=https://releases.linaro.org/14.11/components/toolchain/binutils-linaro/
+  PKG_REV:=2.24.0-2014.11-2
   PKG_SOURCE:=$(PKG_NAME)-linaro-$(PKG_REV).tar.xz
-  PKG_MD5SUM:=8f9b2b2e049d59b1b86ce9657802a353
+  PKG_MD5SUM:=05c1d4bc718ec05f6d53785eea10763f
   BINUTILS_DIR:=$(PKG_NAME)-linaro-$(PKG_REV)
   HOST_BUILD_DIR:=$(BUILD_DIR_TOOLCHAIN)/$(BINUTILS_DIR)
 endif
