diff --git a/miniupnpd/Makefile b/miniupnpd/Makefile
index 1841200..aa8c291 100644
--- a/feeds/routing/miniupnpd/Makefile
+++ b/feeds/routing/miniupnpd/Makefile
@@ -19,6 +19,7 @@ PKG_MAINTAINER:=Markus Stenberg <fingon@iki.fi>
 PKG_LICENSE:=BSD-3-Clause
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/target.mk
 
 define Package/miniupnpd
   SECTION:=net
@@ -53,6 +54,12 @@ endef
 
 define Build/Prepare
 	$(call Build/Prepare/Default)
+	sed -i "s/OpenWRT/`echo $(CONFIG_TARGET_PROFILE) | tr '[:lower:]' '[:upper:]' | cut -c 8-`/" $(PKG_BUILD_DIR)/genconfig.sh
+	$(SED) 's|OS_URL=http://www.openwrt.org/|OS_URL=http://www.tp-link.com/|' $(PKG_BUILD_DIR)/genconfig.sh
+	$(SED) 's|router|Router|' $(PKG_BUILD_DIR)/testupnpdescgen.c
+	$(SED) 's|#define ROOTDEV_MANUFACTURER		OS_NAME|#define ROOTDEV_MANUFACTURER		"TP-LINK"|' $(PKG_BUILD_DIR)/upnpdescstrings.h
+	$(SED) 's|router|Router|' $(PKG_BUILD_DIR)/upnpdescstrings.h
+	$(SED) 's|router|Router|' $(PKG_BUILD_DIR)/upnpglobalvars.c
 	echo "OpenWrt" | tr \(\)\  _ >$(PKG_BUILD_DIR)/os.openwrt
 endef
 
