--- a/include/target.mk
+++ b/include/target.mk
@@ -12,7 +12,14 @@ __target_inc=1
 DEVICE_TYPE?=router
 
 # Default packages - the really basic set
-DEFAULT_PACKAGES:=base-files libc libgcc busybox dropbear mtd uci opkg netifd fstools uclient-fetch logd
+DEFAULT_PACKAGES:=base-files libc libgcc busybox dropbear mtd uci opkg netifd fstools uclient-fetch logd \
+luci luci-app-ddns luci-app-sqm luci-app-upnp luci-proto-relay luci-proto-3g \
+kmod-usb-acm kmod-usb-serial-ipw kmod-usb-serial-option kmod-usb-serial-qualcomm kmod-usb-serial-sierrawireless kmod-usb-serial-wwan \
+kmod-usb-net-cdc-eem kmod-usb-net-cdc-ether kmod-usb-net-cdc-mbim kmod-usb-net-cdc-ncm kmod-usb-net-cdc-subset kmod-usb-net-hso kmod-usb-net-huawei-cdc-ncm kmod-usb-net-kalmia kmod-usb-net-qmi-wwan kmod-usb-net-sierrawireless \
+kmod-sched-cake openvpn-openssl wireguard \
+kmod-sit kmod-ip6-tunnel kmod-nat46 kmod-ppp kmod-mppe kmod-pptp kmod-gre kmod-pppol2tp kmod-l2tp \
+
+
 # For nas targets
 DEFAULT_PACKAGES.nas:=block-mount fdisk lsblk mdadm
 # For router targets
